src/main/scala/reactivemongo/json/collection/jsoncollection.scala

```diff
@@ -186,13 +164,7 @@ object JSONBatchCommands
   implicit object UpsertedReader extends pack.Reader[Upserted] {
     def reads(js: JsValue): JsResult[Upserted] = for {
       ix <- (js \ "index").validate[Int]
-      id <- (js \ "_id") match {
-        case _: JsUndefined =>
-          JsError(__ \ "_id", "error.objectId")
-
-        case js =>
-          BSONFormats.toBSON(js)
-      }
+      id <- (js \ "_id").validate[JsValue].flatMap(BSONFormats.toBSON)
```

Comment:
This used to return JsError in the default case, we are not doing that anymore. is that intentional?